client/src/modules/authorization/components/SignInForm.jsx

```diff
@@ -0,0 +1,219 @@
+import React, { PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import validator from 'validator';
+import { connect } from 'react-redux';
+import _flowRight from 'lodash/flowRight';
+import { Link } from 'react-router-dom';
+
+import AuthInput from './AuthInput';
+import { signIn } from 'modules/authorization/model/actions';
+import PendingButton from 'common/components/PendingButton';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import { UnauthorizedException } from 'common/exceptions/UnauthorizedException';
+import { IntlPropType } from 'common/constants/propTypes';
+
+class SignInForm extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    email: '',
+    isEmailValid: false,
+    isFormValid: false,
+    isPasswordValid: false,
+    password: '',
+    pending: false,
+    signInErrorId: ''
+  };
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onEmailChange = (email, isValid) => {
+    const { signInErrorId } = this.state;
+    const error = isValid ? '' : signInErrorId;
+
+    this.setState({
+      email,
+      isEmailValid: isValid,
+      signInErrorId: error
+    });
+  };
+
+  onPasswordChange = (password, isValid) => {
+    const { signInErrorId } = this.state;
+    const error = isValid ? '' : signInErrorId;
+
+    this.setState(
+      {
+        password,
+        isPasswordValid: isValid,
+        signInErrorId: error
+      },
+      this.comparePasswords
+    );
+  };
+
+  emailValidator = value => {
+    const { isEmail } = validator;
+
+    if (!isEmail(value)) {
+      return 'authorization.input.email.invalid';
+    }
+
+    return '';
+  };
```

Comment:
would be more understandable to set this errors to erros object inside state.

and then in render method destructure those errors from state as pass them to SignUpInput / AuthInput?? 

I think the validator should rather return boolean value or set info about error validation to state. 
Now its kinda hard to understand that validators are returning some strings and these strings are passed down to some other components ETC ETC - it's getting complicated. I think setting this to state and passing from state to a component is more readable.

Reply:
This functionality is used in sign up form already. These strings are not passing down. You pass whole validator function to AuthInput (child component). And AuthInput (child) component uses that validator function and set errors in its state (when of course this is needed). AuthInput is a generic component. Its parent has to pass what kind of input it will be and what validator it should use to check entered value.
Child component uses a validator passed from a parent, so this validator has to inform the child about the message which has to be displayed, otherwise, the child should have hardcode all possible errorIds or there would have to be few very similar input components.